### PR TITLE
#168781673 - Set up docker

### PR DESCRIPTION
**Pull request by Cheza-Dzabala**
#### What does this PR do?

This PR sets up the barefoot nomad project in docker.

#### Description of Task to be completed?

This PR creates a docker image file that allows us to run our application in docker containers.

#### How should this be manually tested?

1. Install Docker on your local machine by following [these](https://docs.docker.com/install/) instructions.

2. After you have ensured that docker is running, clone the repo.

2. Configure your environment with the following as follows:

```
# DATABASE APPLICATION

# DATABASE CONFIG
DB_USERNAME=your_database_username
DB_PASSWORD=your_database_password

# DEVELOPMENT DATABASE
DEVELOPMENT_DB=your_development_database

# TEST DATABASE
TEST_DB=your_test_database

# PRODUCTION DATABASE
PRODUCTION_DB=

# DATABASE HOST // Default to localhost if you are not sure
DATABASE_HOST=database_host

# DOCKER CONFIG

# DOCKER DB CONFIG
DATABASE_PORT=port_to_run_databases

# APPLICATION

APPLICATION_PORT=port_to_run_application

```

3. When the repo is cloned, run the command ` docker-compose build ` to build the image out

4. If you have no errors run ` docker-compose up `. 

5. Access ` http://localhost:<:APPLICATION_PORT>/ ` in Postman, if docker is running correctly, you will be returned with a message:

    ```javascript
   { 
      "status": 200,
      "message": "Welcome to Barefoot Nomad!"
   }
    ```

6. Ensure you have Postgres running 

  _**Note:** Your Postgres Database will run on port ` <:DATABASE_PORT> `. If you're using PgAdmin or Postico ensure you create a database on the server ` localhost:<:DATABASE_PORT> ` 

7. When the server is running, create a database on your new PG server. Ensure your new development database is the same name as your ` DEVELOPMENT_DB `  and your test database is the same name as ` TEST_DB  ` environment variables

8. Run the command ` sequelize db:migrate:all `

9. Run the command sequelize ` db:seed:all `

10. Check your database to ensure proper functionality and correctness. 

#### Any background context you want to provide?

N / A

#### Screenshots (if appropriate)
Screen Shot 2019-10-06 at 21 10 28
Screen Shot 2019-10-06 at 21 10 35
Screen Shot 2019-10-06 at 21 10 45
Screen Shot 2019-10-06 at 21 10 50
Screen Shot 2019-10-06 at 21 10 58

#### Questions:

N / A

### CODE REVIEW

**Review by sabin18 — APPROVED** [review]

**Review by ngireric123 — APPROVED** [review]

**Review by nignanthomas — APPROVED** [review]
Pulled, Tested and after struggling a bit, run the app on docker.
Great job => Approved!

**Review by nakiwuge — CHANGES_REQUESTED** [review]
@Cheza-Dzabala  good job do far. However, you have set up docker for only the development environment please ensure that it is also set up for production. Also, kindly update the readMe with explicit steps of how to run the application with docker

File: package.json [hunk 1]
```diff
@@ -4,9 +4,13 @@
   "description": "A Social platform for the creative at heart",
   "main": "index.js",
   "scripts": {
+    "start:prod": "node --require @babel/register src/index.js",
```
Comment (nakiwuge) [inline package.json]: @Cheza-Dzabala I think this script is not necessary you can use the already existing`production` script.

Comment (Cheza-Dzabala) [reply package.json]: I somehow mussed that script. Apologies.

**Review by nakiwuge — APPROVED** [review]

**Pull request merged**